tests/ord-service/tests/subscription_flow_test.go

```diff
@@ -181,16 +187,18 @@ func TestConsumerProviderFlow(stdT *testing.T) {
 		bundle := fixtures.CreateBundleWithInput(t, ctx, certSecuredGraphQLClient, secondaryTenant, consumerApp.ID, bndlInput)
 		require.NotEmpty(stdT, bundle.ID)
 
+		formationTmplName := "e2e-test-formation-template-name"
+		applicationType := util.ApplicationTypeC4C
+
+		stdT.Logf("Creating formation template for the provider runtime type %q with name %q", conf.SubscriptionProviderAppNameValue, formationTmplName)
+		ft := fixtures.CreateFormationTemplateWithoutInput(stdT, ctx, certSecuredGraphQLClient, formationTmplName, conf.SubscriptionProviderAppNameValue, []string{string(applicationType)}, graphql.ArtifactTypeSubscription)
+		defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, ft.ID)
+
 		consumerFormationName := "consumer-test-scenario"
-		stdT.Logf("Creating formation with name %s...", consumerFormationName)
-		createFormationReq := fixtures.FixCreateFormationRequest(consumerFormationName)
-		executeGQLRequest(stdT, ctx, createFormationReq, consumerFormationName, secondaryTenant)
-		defer func() {
-			stdT.Logf("Deleting formation with name: %s...", consumerFormationName)
-			deleteRequest := fixtures.FixDeleteFormationRequest(consumerFormationName)
-			executeGQLRequest(stdT, ctx, deleteRequest, consumerFormationName, secondaryTenant)
-			stdT.Logf("Successfully deleted formation with name: %s...", consumerFormationName)
-		}()
+		stdT.Logf("Creating formation with name: %q from template with name: %q", consumerFormationName, formationTmplName)
+		formation := fixtures.CreateFormationFromTemplateWithinTenant(stdT, ctx, certSecuredGraphQLClient, secondaryTenant, consumerFormationName, &formationTmplName)
```

Comment:
It didn't work with the old setup (using the default formation template) because after the `subscriptionConsumerSubaccountID` tenant was assigned to formation and then a subscription was created (resulting in a runtimeContext), there was no record in the `labels` table indicating that the runtimeContext is in that formation (`runtime_context_id is not null and key='scenarios'`). This resulted in error in the ORD Service because there it entered the consumer-provider flow -> checked that there is a runtimeContext present -> tries to get all the formations and set them as claims it is part of (none due to the issue) -> and then there is validation to fail if no formations/claims were found.

The reason why no labels with `key='scenario'` were created for the runtimeContext was that there was a mismatch of the runtimeTypes of the default formation Template and the actual one from the test (`conf.SubscriptionProviderAppNameValue`) and then the ASA processing logic was broken.

That part with the default formation template was just not adapted like the other similar places across the tests because with the old ORD flow, all that was needed  was an ASA but now we need the formations as well.

Reply:
I get it, it was a hidden trap :D I was wondering is it a good idea to change the default `conf.SubscriptionProviderAppNameValue` to be equal to the default formation template's runtimeType?

Reply:
Discussed offline - for the time being no adaption is needed, we will create FT and formations in the tests with the proper configuration.